Add reader for EMPAD raw files

sezelt opened the pull request:
This PR adds the ability to read files from the original EMPAD (support for EMPAD 2 is unknown, as I don't have any test files for that). The reader supports reading into memory, memory-mapping, and bin-on-read to RAM. 

Metadata is not supported as there is no meaningful metadata that I am aware of in the file format. (A separate XML file is used for that metadata, but I have not added support for reading from that file. Perhaps this would be handled through a `kwarg` like `metadata_file` if we chose to implement it, since in the examples I've seen the XML does not have the same filename as the data)

bsavitzky commented on py4DSTEM/io/nonnative/empad.py:
```diff
@@ -0,0 +1,91 @@
+from ..datastructure import DataCube
+from ...process.utils import bin2D, tqdmnd
+
+from pdb import set_trace
```
We can remove this, no?

bsavitzky commented on py4DSTEM/io/nonnative/empad.py:
```diff
@@ -0,0 +1,91 @@
+                    raw data only can be accessed data[:,:,:128,:]
+    """
+
+    # fmt: off
```
rm? This is leftover from using Black for formatting?

bsavitzky commented on py4DSTEM/io/nonnative/empad.py:
```diff
@@ -0,0 +1,91 @@
+    assert(isinstance(binfactor, int)), "Error: argument binfactor must be an integer"
+    assert(binfactor >= 1), "Error: binfactor must be >= 1"
+    assert(metadata is False), "Error: EMPAD Reader does not support metadata."
+    # fmt: on
```
rm?

bsavitzky commented on py4DSTEM/io/nonnative/empad.py:
```diff
@@ -0,0 +1,91 @@
+    shape1 = imLast["metadata"][0][128 + 12 : 128 + 16]
+    # kShape = shape0[2:4]  # detector shape (should always be row x col)
+    rShape = 1 + shape1[0:2] - shape0[0:2]  # scan shape
+
```
rm blank line

bsavitzky commented on py4DSTEM/io/nonnative/empad.py:
```diff
@@ -0,0 +1,91 @@
+    By extracting the scan position of the first and last frames, the function determines the scan
+    size. Then, the full dataset is loaded.
+
+    Note that the output of this function is a datacube which includes the 2 rows of metadata.
+
+    Accepts:
+        filename    (str) path to the empad file
+
+    Returns:
+        data        (ndarray) the 4D datacube, including the metadata.
+                    raw data only can be accessed data[:,:,:128,:]
```
It says here that the fn returns the datacube *with* the 2 metadata lines, but the code (lines 64,68,72) do *not* appear to be including the metadata lines in the returned datacube.  I think this is probably the right way to do this, but we should then change the `Note` and `Returns` descriptions in this docstring

bsavitzky reviewed (changes requested):
Looks good.  The only meaningful edit requested is to please confirm that this function does *not* include the 2 metadata lines in the output datacube, and to edit the docstring accordingly

sezelt commented:
@bsavitzky I agree with all your comments, and addressed them in this new commit. I edited the docstring to clarify that we do indeed return only the 128x128 valid data region, and also corrected that we are returning the `DataCube`, not a `np.ndarray`. Those other little comment lines were indeed there to prevent black from reformatting those assert statements to look insane, but since we in general aren't (yet!) running a formatter on the whole codebase I just nixed them.

bsavitzky commented:
> @bsavitzky I agree with all your comments, and addressed them in this new commit. I edited the docstring to clarify that we do indeed return only the 128x128 valid data region, and also corrected that we are returning the `DataCube`, not a `np.ndarray`. Those other little comment lines were indeed there to prevent black from reformatting those assert statements to look insane, but since we in general aren't (yet!) running a formatter on the whole codebase I just nixed them.

Thanks @sezelt - one last thing - I take it you don't still need to be importing the python debugger, yes?  Cool to remove this line?

sezelt commented:
Yes, it's safe to remove `pdb`.
